src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -981,9 +981,7 @@ public function createService(Definition $definition, $id, $tryProxy = true)
         }
 
         if ($callable = $definition->getConfigurator()) {
-            if (is_array($callable)) {
-                $callable[0] = $callable[0] instanceof Reference ? $this->get((string) $callable[0]) : $parameterBag->resolveValue($callable[0]);
-            }
+            $callable = $this->resolveServices($parameterBag->resolveValue($callable));
```

Comment:
Yes, but that means we still have duplicated and potentially buggy code. If somebody adds functionality to resolveServices(), they need to think about changing this code too (like happened before). IMO this should be fixed.